### PR TITLE
Hw marc 2

### PR DESCRIPTION
**Pull request by Zmetser**
Here's our homework. Same as last, we didn't bother with tests or to make it actually run.

- [x] move the statistic calculation logic into a separate IMT
- [x] consume a queue, and calculate that stat by queue messages (implemented in the IMT)
- [x] create a "process" file where you can use the IMT

@golya

### CODE REVIEW

File: src/server/internal/stats/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+function Stats(cache, requestMonitor, queue) {
```
Comment (Zmetser) [inline src/server/internal/stats/index.js]: @golya `cache` and `queue` are `undefined` here, while `requestMonitor` is not. What magic do I have to do to make them injected?

**Comment by Zmetser** [comment]
@golya I still can't run the server by `npm run start`. I get an error, and localhost:8080 doesn't load. Why? What is wrong?

```bash
➜  greenfox-mc git:(master) npm run start

> greenfox-smc@1.0.0 start /Users/okovacs/Development/greenfox/greenfox-mc
> PUBLIC_DIR='../client/public' nodemon src/server/server.js --ignore dist --ignore node_modules --exec babel-node

[nodemon] 1.11.0
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `babel-node src/server/server.js`
WARNING: No configurations found in configuration directory:/Users/okovacs/Development/greenfox/greenfox-mc/config
WARNING: To disable this warning set SUPPRESS_NO_CONFIG_WARNING in the environment.
Server listening on port 8080!
(node:66999) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'type' of undefined
```

File: src/server/internal/stats/index.js (hunk 1 above)
Comment (golya) [inline src/server/internal/stats/index.js]: Do you have config file?

Comment (Zmetser) [inline src/server/internal/stats/index.js]: No, I only have what's in the repo.

File: src/server/internal/requestmonitor/index.js [hunk 2]
```diff
@@ -12,6 +11,8 @@ function RequestMonitor(store) {
       }`,
       {url}
     );
+
+    queue.publish('stats', 'registerIncomingRequest');
```
Comment (Zmetser) [inline src/server/internal/requestmonitor/index.js]: That was the fastest solution. How Would You did it?

File: src/server/internal/stats/index.js [hunk 3]
```diff
@@ -1,5 +1,13 @@
 
-function Stats(cache, requestMonitor) {
+function Stats(cache, requestMonitor, queue) {
+
+  queue.consume('stats', (message) => {
+    if (message === 'registerIncomingRequest') {
```
Comment (Zmetser) [inline src/server/internal/stats/index.js]: What can I say, I'm a rebel!

File: src/server/internal/stats/index.js (hunk 1 above)
Comment (golya) [inline src/server/internal/stats/index.js]: You can copy the config/default.example.json to config/default.json and it should work.

**Pull request closed without merging**